EasyMonads.Test/MaybeTests/EquatableTests.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections.Generic;
+using NUnit.Framework;
+using static EasyMonads.Core;
+
+namespace EasyMonads.Test.MaybeTests
+{
+   [TestFixture]
+   internal class EquatableTests
+   {
+      [Test]
+      public void Maybe_Equality_Test()
+      {
+         var someA = Maybe(42);
+         var someB = Some(42);
+         var none = Maybe<int>.None;
+
+
+         Assert.True(someA.Equals(someB));
+         Assert.True(someA == someB);
+         Assert.True(none == None);
+      }
+
+      [Test]
+      public void Maybe_Equality_Test_When_Implicitly_Converted()
+      {
+         if (Some(42) == 42)
+         {
+            Assert.Pass();
+         }
+         else
+         {
+            Assert.Fail();
+         }
+      }
+
+      [Test]
+      public void Maybe_Inequality_Test()
+      {
+         var maybeA = Maybe(42);
+         var maybeB = Some(12);
+         Maybe<int> none = None;
+         var someObj = Some(new object());
+
+         Assert.False(maybeA.Equals(maybeB));
+         Assert.True(maybeA != maybeB);
+         Assert.True(maybeA != none);
+         
+         Assert.True(maybeA != someObj);
+         Assert.False(maybeA == someObj);
+      }
+
+      [Test]
+      public void Maybe_Inequality_Test_When_Implicitly_Converted()
+      {
+         if (Some(42) != 12)
+         {
+            Assert.Pass();
+         }
+         else
+         {
+            Assert.Fail();
+         }
+      }
+
+      [Test]
+      public void Maybe_is_Truthy_or_Falsy()
+      {
+         var someInt = Maybe<int?>.FromNullable(42);
+         var noneInt = Maybe<int?>.FromNullable(null);

```

Comment:
This case bugs me.  More of me leans towards this instance actually being in the `Some` state.  Considering we may have gotten a result from somewhere and the value of that result is actually `null`.  Wouldn't want the caller to see the `None` and proceed as if we failed to get a result.

To clear up this kind of ambiguity I usually use an Either, but could be the same problem exists with the Either falling into the `Neither` state.

Not sure that I really took a hard line one way or the other in the code previously.

Reply:
It bugs me too. Changing the behavior would be a breaking change. If you open an issue we can discuss options. (yeah, pun intended!)

Reply:
#16